diff --git app/code/core/Enterprise/Index/Model/Resource/Lock/Resource.php app/code/core/Enterprise/Index/Model/Resource/Lock/Resource.php
index b91860e..d79a66c 100644
--- app/code/core/Enterprise/Index/Model/Resource/Lock/Resource.php
+++ app/code/core/Enterprise/Index/Model/Resource/Lock/Resource.php
@@ -34,6 +34,14 @@
 class Enterprise_Index_Model_Resource_Lock_Resource extends Mage_Core_Model_Resource
 {
     /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->_connections = Mage::getSingleton('core/resource')->getConnections();
+    }
+
+    /**
      * Creates a connection to resource whenever needed
      *
      * @param string $name
@@ -68,10 +76,21 @@ class Enterprise_Index_Model_Resource_Lock_Resource extends Mage_Core_Model_Reso
             return $this->_connections[$origName];
         }
 
+        $origConfigParams = $connConfig->asArray();
         if ($extendConfigWith) {
             $connConfig->extend(Mage::getConfig()->getResourceConnectionConfig($extendConfigWith), true);
         }
 
+        $configDiff = array_diff_assoc($connConfig->asArray(), $origConfigParams);
+        if (!$configDiff) {
+            $index = $name;
+            $origName = $connConfig->getParent()->getName();
+            if (isset($this->_connections[$origName])) {
+                $this->_connections[$index] = $this->_connections[$origName];
+                return $this->_connections[$origName];
+            }
+        }
+
         $connection = $this->_newConnection((string)$connConfig->type, $connConfig);
         if ($connection) {
             if (Mage::app()->getIsCacheLocked()) {
diff --git app/code/core/Mage/Core/Model/Resource.php app/code/core/Mage/Core/Model/Resource.php
index d96ec1c..d8952e9 100644
--- app/code/core/Mage/Core/Model/Resource.php
+++ app/code/core/Mage/Core/Model/Resource.php
@@ -125,6 +125,16 @@ class Mage_Core_Model_Resource
     }
 
     /**
+     * Get Instances of actual connections
+     *
+     * @return array
+     */
+    public function getConnections()
+    {
+        return $this->_connections;
+    }
+
+    /**
      * Retrieve connection adapter class name by connection type
      *
      * @param string $type  the connection type
diff --git app/etc/config.xml app/etc/config.xml
index 3b90ef7..49ceeee 100644
--- app/etc/config.xml
+++ app/etc/config.xml
@@ -41,6 +41,7 @@
                     <initStatements>SET NAMES utf8</initStatements>
                     <type>pdo_mysql</type>
                     <active>0</active>
+                    <persistent>0</persistent>
                 </connection>
             </default_setup>
             <default_write>
