diff --git lib/Varien/Cache/Backend/Database.php lib/Varien/Cache/Backend/Database.php
index b8719f5..2127abf 100644
--- lib/Varien/Cache/Backend/Database.php
+++ lib/Varien/Cache/Backend/Database.php
@@ -137,18 +137,19 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
      */
     public function load($id, $doNotTestCacheValidity = false)
     {
-        if ($this->_options['store_data']) {
-            $select = $this->_getAdapter()->select()
-                ->from($this->_getDataTable(), 'data')
-                ->where('id=:cache_id');
-
-            if (!$doNotTestCacheValidity) {
-                $select->where('expire_time=0 OR expire_time>?', time());
-            }
-            return $this->_getAdapter()->fetchOne($select, array('cache_id'=>$id));
-        } else {
+        if (!$this->_options['store_data']) {
             return false;
         }
+
+        $select = $this->_getAdapter()->select()
+            ->from($this->_getDataTable(), 'data')
+            ->where('id=:cache_id');
+
+        if (!$doNotTestCacheValidity) {
+            $select->where('expire_time = 0 OR expire_time > ?', time());
+        }
+
+        return $this->_getAdapter()->fetchOne($select, array('cache_id' => $id));
     }
 
     /**
@@ -159,31 +160,34 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
      */
     public function test($id)
     {
-        if ($this->_options['store_data']) {
-            $select = $this->_getAdapter()->select()
-                ->from($this->_getDataTable(), 'update_time')
-                ->where('id=:cache_id')
-                ->where('expire_time=0 OR expire_time>?', time());
-            return $this->_getAdapter()->fetchOne($select, array('cache_id'=>$id));
-        } else {
+        if (!$this->_options['store_data']) {
             return false;
         }
+
+        $select = $this->_getAdapter()->select()
+            ->from($this->_getDataTable(), 'update_time')
+            ->where('id=:cache_id')
+            ->where('expire_time = 0 OR expire_time > ?', time());
+        return $this->_getAdapter()->fetchOne($select, array('cache_id' => $id));
     }
 
     /**
-     * Save some string datas into a cache record
+     * Save data into a cache storage
      *
      * Note : $data is always "string" (serialization is done by the
      * core not by the backend)
      *
-     * @param  string $data            Datas to cache
-     * @param  string $id              Cache id
-     * @param  array $tags             Array of strings, the cache record will be tagged by each string entry
-     * @param  int   $specificLifetime If != false, set a specific lifetime for this cache record (null => infinite lifetime)
-     * @return boolean true if no problem
+     * @param  string $data Data to cache
+     * @param  string $id Cache id
+     * @param  array $tags Array of strings, the cache record will be tagged by each string entry
+     * @param  int|bool|null $specificLifetime If != false, set a specific lifetime for this cache record
+     *                                    (null => infinite lifetime)
+     *
+     * @return bool true if no problem
      */
     public function save($data, $id, $tags = array(), $specificLifetime = false)
     {
+        $result = true;
         if ($this->_options['store_data']) {
             $adapter    = $this->_getAdapter();
             $dataTable  = $this->_getDataTable();
@@ -209,8 +213,8 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
                 return false;
             }
         }
-        $tagRes = $this->_saveTags($id, $tags);
-        return $tagRes;
+
+        return $result && $this->_saveTags($id, $tags);
     }
 
     /**
@@ -221,13 +225,13 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
      */
     public function remove($id)
     {
+        $adapter = $this->_getAdapter();
+        $result = true;
         if ($this->_options['store_data']) {
-            $adapter = $this->_getAdapter();
-            $result = $adapter->delete($this->_getDataTable(), array('id=?'=>$id));
-            return $result;
-        } else {
-            return false;
+            $result = $adapter->delete($this->_getDataTable(), array('id = ?' => $id));
         }
+
+        return $result && $adapter->delete($this->_getTagsTable(), array('cache_id = ?' => $id));
     }
 
     /**
@@ -250,23 +254,18 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
     public function clean($mode = Zend_Cache::CLEANING_MODE_ALL, $tags = array())
     {
         $adapter = $this->_getAdapter();
+        $result = true;
+
         switch($mode) {
             case Zend_Cache::CLEANING_MODE_ALL:
                 if ($this->_options['store_data']) {
-                    $result = $adapter->query('TRUNCATE TABLE '.$this->_getDataTable());
-                } else {
-                    $result = true;
+                    $result = $adapter->query('TRUNCATE TABLE ' . $this->_getDataTable());
                 }
-                $result = $result && $adapter->query('TRUNCATE TABLE '.$this->_getTagsTable());
+                $result = $result && $adapter->query('TRUNCATE TABLE ' . $this->_getTagsTable());
                 break;
             case Zend_Cache::CLEANING_MODE_OLD:
                 if ($this->_options['store_data']) {
-                    $result = $adapter->delete($this->_getDataTable(), array(
-                        'expire_time> ?' => 0,
-                        'expire_time<= ?' => time()
-                    ));
-                } else {
-                    $result = true;
+                    $result = $this->_cleanOldCache();
                 }
                 break;
             case Zend_Cache::CLEANING_MODE_MATCHING_TAG:
@@ -283,6 +282,52 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
     }
 
     /**
+     * Clean old cache data and related cache tag data
+     *
+     * @return bool
+     */
+    protected function _cleanOldCache()
+    {
+        $time = time();
+        $adapter = $this->_getAdapter();
+
+        $select = $adapter->select()
+            ->from($this->_getDataTable(), 'id')
+            ->where('expire_time > ?', 0)
+            ->where('expire_time <= ?', $time);
+        $statement = $adapter->query($select);
+
+        $cacheIdsToRemove = array();
+        $counter = 0;
+        $result = true;
+        while (($row = $statement->fetch()) == true) {
+            $cacheIdsToRemove[] = $row['id'];
+            $counter++;
+            if ($counter > 100) {
+                $result = $result && $adapter->delete(
+                    $this->_getTagsTable(),
+                    array('cache_id IN (?)' => $cacheIdsToRemove)
+                );
+                $cacheIdsToRemove = array();
+                $counter = 0;
+            }
+        }
+        if (!empty($cacheIdsToRemove)) {
+            $result = $result && $adapter->delete(
+                $this->_getTagsTable(),
+                array('cache_id IN (?)' => $cacheIdsToRemove)
+            );
+        }
+
+        $result = $result && $adapter->delete($this->_getDataTable(), array(
+            'expire_time > ?' => 0,
+            'expire_time <= ?' => $time
+        ));
+
+        return $result;
+    }
+
+    /**
      * Return an array of stored cache ids
      *
      * @return array array of stored cache ids (string)
@@ -412,15 +457,15 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
      */
     public function touch($id, $extraLifetime)
     {
-        if ($this->_options['store_data']) {
-            return $this->_getAdapter()->update(
-                $this->_getDataTable(),
-                array('expire_time'=>new Zend_Db_Expr('expire_time+'.$extraLifetime)),
-                array('id=?'=>$id, 'expire_time = 0 OR expire_time>'=>time())
-            );
-        } else {
+        if (!$this->_options['store_data']) {
             return true;
         }
+
+        return $this->_getAdapter()->update(
+            $this->_getDataTable(),
+            array('expire_time' => new Zend_Db_Expr('expire_time + ' . $extraLifetime)),
+            array('id = ?' => $id, 'expire_time = 0 OR expire_time > ? ' => time())
+        );
     }
 
     /**
@@ -472,6 +517,7 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
             ->where('cache_id=?', $id)
             ->where('tag IN(?)', $tags);
 
+        $result = true;
         $existingTags = $adapter->fetchCol($select);
         $insertTags = array_diff($tags, $existingTags);
         if (!empty($insertTags)) {
@@ -483,10 +529,10 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
                 $bind[] = $tag;
                 $bind[] = $id;
             }
-            $query.= implode(',', $lines);
-            $adapter->query($query, $bind);
+            $query .= implode(',', $lines);
+            $result = $adapter->query($query, $bind);
         }
-        $result = true;
+
         return $result;
     }
 
@@ -499,46 +545,58 @@ class Varien_Cache_Backend_Database extends Zend_Cache_Backend implements Zend_C
      */
     protected function _cleanByTags($mode, $tags)
     {
-        if ($this->_options['store_data']) {
-            $adapter = $this->_getAdapter();
-            $select = $adapter->select()
-                ->from($this->_getTagsTable(), 'cache_id');
-            switch ($mode) {
-                case Zend_Cache::CLEANING_MODE_MATCHING_TAG:
-                    $select->where('tag IN (?)', $tags)
-                        ->group('cache_id')
-                        ->having('COUNT(cache_id)='.count($tags));
+        $adapter = $this->_getAdapter();
+        $result = true;
+        $select = $adapter->select()
+            ->from($this->_getTagsTable(), 'cache_id');
+        switch ($mode) {
+            case Zend_Cache::CLEANING_MODE_MATCHING_TAG:
+                $select->where('tag IN (?)', $tags)
+                    ->group('cache_id')
+                    ->having('COUNT(cache_id) = ' . count($tags));
                 break;
-                case Zend_Cache::CLEANING_MODE_NOT_MATCHING_TAG:
-                    $select->where('tag NOT IN (?)', $tags);
+            case Zend_Cache::CLEANING_MODE_NOT_MATCHING_TAG:
+                $select->where('tag NOT IN (?)', $tags);
                 break;
-                case Zend_Cache::CLEANING_MODE_MATCHING_ANY_TAG:
-                    $select->where('tag IN (?)', $tags);
+            case Zend_Cache::CLEANING_MODE_MATCHING_ANY_TAG:
+                $select->where('tag IN (?)', $tags);
                 break;
-                default:
-                    Zend_Cache::throwException('Invalid mode for _cleanByTags() method');
+            default:
+                Zend_Cache::throwException('Invalid mode for _cleanByTags() method');
                 break;
-            }
+        }
 
-            $result = true;
-            $ids = array();
-            $counter = 0;
-            $stmt = $adapter->query($select);
-            while ($row = $stmt->fetch()) {
-                $ids[] = $row['cache_id'];
-                $counter++;
-                if ($counter>100) {
-                    $result = $result && $adapter->delete($this->_getDataTable(), array('id IN (?)' => $ids));
-                    $ids = array();
-                    $counter = 0;
+        $cacheIdsToRemove = array();
+        $counter = 0;
+        $statement = $adapter->query($select);
+        while (($row = $statement->fetch()) == true) {
+            $cacheIdsToRemove[] = $row['cache_id'];
+            $counter++;
+            if ($counter > 100) {
+                if ($this->_options['store_data']) {
+                    $result = $result && $adapter->delete(
+                        $this->_getDataTable(),
+                        array('id IN (?)' => $cacheIdsToRemove)
+                    );
                 }
+                $result = $result && $adapter->delete(
+                    $this->_getTagsTable(),
+                    array('cache_id IN (?)' => $cacheIdsToRemove)
+                );
+                $cacheIdsToRemove = array();
+                $counter = 0;
             }
-            if (!empty($ids)) {
-                $result = $result && $adapter->delete($this->_getDataTable(), array('id IN (?)' => $ids));
+        }
+        if (!empty($cacheIdsToRemove)) {
+            if ($this->_options['store_data']) {
+                $result = $result && $adapter->delete(
+                    $this->_getDataTable(), array('id IN (?)' => $cacheIdsToRemove)
+                );
             }
-            return $result;
-        } else {
-            return true;
+            $result = $result && $adapter->delete(
+                $this->_getTagsTable(), array('cache_id IN (?)' => $cacheIdsToRemove)
+            );
         }
+        return $result;
     }
 }
