diff --git app/code/core/Mage/Catalog/etc/config.xml app/code/core/Mage/Catalog/etc/config.xml
index db1227c..9abcf15 100644
--- app/code/core/Mage/Catalog/etc/config.xml
+++ app/code/core/Mage/Catalog/etc/config.xml
@@ -28,7 +28,7 @@
 <config>
     <modules>
         <Mage_Catalog>
-            <version>1.6.0.0.18</version>
+            <version>1.6.0.0.18.1.2</version>
         </Mage_Catalog>
     </modules>
     <admin>
diff --git app/code/core/Mage/Catalog/sql/catalog_setup/upgrade-1.6.0.0.18.1.1-1.6.0.0.18.1.2.php app/code/core/Mage/Catalog/sql/catalog_setup/upgrade-1.6.0.0.18.1.1-1.6.0.0.18.1.2.php
new file mode 100644
index 0000000..f284ea1
--- /dev/null
+++ app/code/core/Mage/Catalog/sql/catalog_setup/upgrade-1.6.0.0.18.1.1-1.6.0.0.18.1.2.php
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Magento Enterprise Edition
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Magento Enterprise Edition License
+ * that is bundled with this package in the file LICENSE_EE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://www.magentocommerce.com/license/enterprise-edition
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magentocommerce.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Magento to newer
+ * versions in the future. If you wish to customize Magento for your
+ * needs please refer to http://www.magentocommerce.com for more information.
+ *
+ * @category    Mage
+ * @package     Mage_Catalog
+ * @copyright   Copyright (c) 2013 Magento Inc. (http://www.magentocommerce.com)
+ * @license     http://www.magentocommerce.com/license/enterprise-edition
+ */
+
+/** @var $installer Mage_Catalog_Model_Resource_Setup */
+$installer  = $this;
+/**
+ * Create table 'catalog/category_product_indexer_tmp'
+ */
+if (!$installer->tableExists('catalog/category_product_indexer_tmp')) {
+    $table = $installer->getConnection()
+        ->newTable($installer->getTable('catalog/category_product_indexer_tmp'))
+        ->addColumn('category_id', Varien_Db_Ddl_Table::TYPE_INTEGER, null, array(
+            'unsigned'  => true,
+            'nullable'  => false,
+            'default'   => '0',
+        ), 'Category ID')
+        ->addColumn('product_id', Varien_Db_Ddl_Table::TYPE_INTEGER, null, array(
+            'unsigned'  => true,
+            'nullable'  => false,
+            'default'   => '0',
+        ), 'Product ID')
+        ->addColumn('position', Varien_Db_Ddl_Table::TYPE_INTEGER, null, array(
+            'nullable'  => false,
+            'default'   => '0',
+        ), 'Position')
+        ->addColumn('is_parent', Varien_Db_Ddl_Table::TYPE_SMALLINT, null, array(
+            'unsigned'  => true,
+            'nullable'  => false,
+            'default'   => '0',
+        ), 'Is Parent')
+        ->addColumn('store_id', Varien_Db_Ddl_Table::TYPE_SMALLINT, null, array(
+            'unsigned'  => true,
+            'nullable'  => false,
+            'default'   => '0',
+        ), 'Store ID')
+        ->addColumn('visibility', Varien_Db_Ddl_Table::TYPE_SMALLINT, null, array(
+            'unsigned'  => true,
+            'nullable'  => false,
+        ), 'Visibility')
+        ->setComment('Catalog Category Product Indexer Temp Table')
+        ->setOption('type', Varien_Db_Adapter_Pdo_Mysql::ENGINE_MEMORY);
+    $installer->getConnection()->createTable($table);
+}
