diff --git app/code/core/Enterprise/PageCache/Model/Processor.php app/code/core/Enterprise/PageCache/Model/Processor.php
index 87b773a..472ae81 100644
--- app/code/core/Enterprise/PageCache/Model/Processor.php
+++ app/code/core/Enterprise/PageCache/Model/Processor.php
@@ -720,6 +720,11 @@ class Enterprise_PageCache_Model_Processor
          * Define server HTTP HOST
          */
         if (isset($_SERVER['HTTP_HOST'])) {
+            if (strpos($_SERVER['HTTP_HOST'], ',') !== false || strpos($_SERVER['HTTP_HOST'], ';') !== false) {
+                $response = new Zend_Controller_Response_Http();
+                $response->setHttpResponseCode(400)->sendHeaders();
+                exit();
+            }
             $uri = $_SERVER['HTTP_HOST'];
         } elseif (isset($_SERVER['SERVER_NAME'])) {
             $uri = $_SERVER['SERVER_NAME'];
diff --git app/code/core/Mage/Api/Model/Server/Adapter/Soap.php app/code/core/Mage/Api/Model/Server/Adapter/Soap.php
index a98ef58..3bcba1b 100644
--- app/code/core/Mage/Api/Model/Server/Adapter/Soap.php
+++ app/code/core/Mage/Api/Model/Server/Adapter/Soap.php
@@ -233,9 +233,9 @@ class Mage_Api_Model_Server_Adapter_Soap
             : $urlModel->getUrl('*/*/*');
 
         if ( $withAuth ) {
-            $phpAuthUser = $this->getController()->getRequest()->getServer('PHP_AUTH_USER', false);
-            $phpAuthPw = $this->getController()->getRequest()->getServer('PHP_AUTH_PW', false);
-            $scheme = $this->getController()->getRequest()->getScheme();
+            $phpAuthUser = rawurlencode($this->getController()->getRequest()->getServer('PHP_AUTH_USER', false));
+            $phpAuthPw = rawurlencode($this->getController()->getRequest()->getServer('PHP_AUTH_PW', false));
+            $scheme = rawurlencode($this->getController()->getRequest()->getScheme());
 
             if ($phpAuthUser && $phpAuthPw) {
                 $wsdlUrl = sprintf("%s://%s:%s@%s", $scheme, $phpAuthUser, $phpAuthPw,
diff --git app/code/core/Mage/Catalog/Model/Product/Api/V2.php app/code/core/Mage/Catalog/Model/Product/Api/V2.php
index 2d14c83..1b6db79 100644
--- app/code/core/Mage/Catalog/Model/Product/Api/V2.php
+++ app/code/core/Mage/Catalog/Model/Product/Api/V2.php
@@ -108,7 +108,7 @@ class Mage_Catalog_Model_Product_Api_V2 extends Mage_Catalog_Model_Product_Api
      */
     public function create($type, $set, $sku, $productData, $store = null)
     {
-        if (!$type || !$set || !$sku) {
+        if (!$type || !$set || !$sku || !is_object($productData)) {
             $this->_fault('data_invalid');
         }
 
@@ -243,6 +243,9 @@ class Mage_Catalog_Model_Product_Api_V2 extends Mage_Catalog_Model_Product_Api
      */
     protected function _prepareDataForSave ($product, $productData)
     {
+        if (!is_object($productData)) {
+            $this->_fault('data_invalid');
+        }
         if (property_exists($productData, 'website_ids') && is_array($productData->website_ids)) {
             $product->setWebsiteIds($productData->website_ids);
         }
diff --git app/code/core/Mage/Core/Controller/Request/Http.php app/code/core/Mage/Core/Controller/Request/Http.php
index 3370959..08b7a15 100644
--- app/code/core/Mage/Core/Controller/Request/Http.php
+++ app/code/core/Mage/Core/Controller/Request/Http.php
@@ -298,11 +298,19 @@ class Mage_Core_Controller_Request_Http extends Zend_Controller_Request_Http
         if (!isset($_SERVER['HTTP_HOST'])) {
             return false;
         }
+        $host = $_SERVER['HTTP_HOST'];
         if ($trimPort) {
-            $host = explode(':', $_SERVER['HTTP_HOST']);
-            return $host[0];
+            $hostParts = explode(':', $_SERVER['HTTP_HOST']);
+            $host =  $hostParts[0];
         }
-        return $_SERVER['HTTP_HOST'];
+
+        if (strpos($host, ',') !== false || strpos($host, ';') !== false) {
+            $response = new Zend_Controller_Response_Http();
+            $response->setHttpResponseCode(400)->sendHeaders();
+            exit();
+        }
+
+        return $host;
     }
 
     /**
diff --git app/design/frontend/base/default/template/page/js/cookie.phtml app/design/frontend/base/default/template/page/js/cookie.phtml
index 1413438..708ac7e 100644
--- app/design/frontend/base/default/template/page/js/cookie.phtml
+++ app/design/frontend/base/default/template/page/js/cookie.phtml
@@ -34,7 +34,7 @@
 
 <script type="text/javascript">
 //<![CDATA[
-Mage.Cookies.path     = '<?php echo $this->getPath()?>';
-Mage.Cookies.domain   = '<?php echo $this->getDomain()?>';
+Mage.Cookies.path     = '<?php echo Mage::helper('core')->jsQuoteEscape($this->getPath()) ?>';
+Mage.Cookies.domain   = '<?php echo Mage::helper('core')->jsQuoteEscape($this->getDomain()) ?>';
 //]]>
 </script>
diff --git app/design/frontend/enterprise/default/template/giftregistry/search/form.phtml app/design/frontend/enterprise/default/template/giftregistry/search/form.phtml
index 5dfdae8..effafd6 100644
--- app/design/frontend/enterprise/default/template/giftregistry/search/form.phtml
+++ app/design/frontend/enterprise/default/template/giftregistry/search/form.phtml
@@ -138,7 +138,7 @@ $('params_type_id').observe('change', advancedFormUpdate);
 
 <?php if ($this->getFormData('type_id')): ?>
     $A($('params_type_id').options).each(function(option){
-        if (option.value==<?php echo $this->getFormData('type_id') ?>) option.selected = true;
+        if (option.value==<?php echo (int)$this->getFormData('type_id') ?>) option.selected = true;
     });
     advancedFormUpdate();
 <?php endif; ?>
diff --git app/design/frontend/rwd/enterprise/template/giftregistry/search/form.phtml app/design/frontend/rwd/enterprise/template/giftregistry/search/form.phtml
index def80da..b41b71c 100644
--- app/design/frontend/rwd/enterprise/template/giftregistry/search/form.phtml
+++ app/design/frontend/rwd/enterprise/template/giftregistry/search/form.phtml
@@ -138,7 +138,7 @@ $('params_type_id').observe('change', advancedFormUpdate);
 
 <?php if ($this->getFormData('type_id')): ?>
     $A($('params_type_id').options).each(function(option){
-        if (option.value==<?php echo $this->getFormData('type_id') ?>) option.selected = true;
+        if (option.value==<?php echo (int)$this->getFormData('type_id') ?>) option.selected = true;
     });
     advancedFormUpdate();
 <?php endif; ?>
