diff --git app/code/core/Mage/Checkout/controllers/CartController.php app/code/core/Mage/Checkout/controllers/CartController.php
index 3a9bb88..2425152 100644
--- app/code/core/Mage/Checkout/controllers/CartController.php
+++ app/code/core/Mage/Checkout/controllers/CartController.php
@@ -594,11 +594,13 @@ class Mage_Checkout_CartController extends Mage_Core_Controller_Front_Action
         try {
             $codeLength = strlen($couponCode);
             $isCodeLengthValid = $codeLength && $codeLength <= Mage_Checkout_Helper_Cart::COUPON_CODE_MAX_LENGTH;
+            $newCouponCode = $isCodeLengthValid ? $couponCode : '';
 
             $this->_getQuote()->getShippingAddress()->setCollectShippingRates(true);
-            $this->_getQuote()->setCouponCode($isCodeLengthValid ? $couponCode : '')
+            $this->_getQuote()->setCouponCode($newCouponCode)
                 ->collectTotals()
                 ->save();
+            $this->_getSession()->setCartCouponCode($newCouponCode);
 
             if ($codeLength) {
                 if ($isCodeLengthValid && $couponCode == $this->_getQuote()->getCouponCode()) {
