diff --git app/code/core/Mage/Adminhtml/Block/Widget/Form.php app/code/core/Mage/Adminhtml/Block/Widget/Form.php
index 3957d1296e6..57dc736b434 100644
--- app/code/core/Mage/Adminhtml/Block/Widget/Form.php
+++ app/code/core/Mage/Adminhtml/Block/Widget/Form.php
@@ -190,7 +190,7 @@ class Mage_Adminhtml_Block_Widget_Form extends Mage_Adminhtml_Block_Widget
                         'label'     => $attribute->getFrontend()->getLabel(),
                         'class'     => $attribute->getFrontend()->getClass(),
                         'required'  => $attribute->getIsRequired(),
-                        'note'      => $attribute->getNote(),
+                        'note'      => $this->escapeHtml($attribute->getNote()),
                     )
                 )
                 ->setEntityAttribute($attribute);
diff --git app/code/core/Mage/Adminhtml/controllers/Catalog/Product/AttributeController.php app/code/core/Mage/Adminhtml/controllers/Catalog/Product/AttributeController.php
index 6e85ed8c3c0..dcbd47c9564 100644
--- app/code/core/Mage/Adminhtml/controllers/Catalog/Product/AttributeController.php
+++ app/code/core/Mage/Adminhtml/controllers/Catalog/Product/AttributeController.php
@@ -267,6 +267,7 @@ class Mage_Adminhtml_Catalog_Product_AttributeController extends Mage_Adminhtml_
             if(!isset($data['apply_to'])) {
                 $data['apply_to'] = array();
             }
+            $data['entity_type_id'] = $model->getEntityTypeId();
 
             //filter
             $data = $this->_filterPostData($data);
diff --git app/code/core/Mage/Eav/Model/Attribute.php app/code/core/Mage/Eav/Model/Attribute.php
index aeb5c6a77a8..a7e948bbcb4 100644
--- app/code/core/Mage/Eav/Model/Attribute.php
+++ app/code/core/Mage/Eav/Model/Attribute.php
@@ -116,7 +116,7 @@ abstract class Mage_Eav_Model_Attribute extends Mage_Eav_Model_Entity_Attribute
         if (is_array($rules)) {
             return $rules;
         } else if (!empty($rules)) {
-            return unserialize($rules);
+            return Mage::helper('core/unserializeArray')->unserialize($rules);
         }
         return array();
     }
